src/fragmentarium/ui/info/genres.ts

```diff
@@ -0,0 +1,117 @@
+import _ from 'lodash'
+
+export interface Tree {
+  value: string
+  nodes?: ReadonlyArray<Tree>
+}
+export function flattenTree(genres: Tree) {
+  const flattenedTree: string[][] = []
+  let stack: string[] = []
+  const visited: Set<string> = new Set()
+  recursiveDepthSearch(genres)
+
+  function recursiveDepthSearch(tree: Tree) {
+    stack.push(tree.value)
+    if (!visited.has(genres.value)) {
+      recursiveDepthSearchHelper(tree)
+    }
+  }
+
+  function recursiveDepthSearchHelper(tree: Tree) {
```

Comment:
The function name should tell what it does and usually be a verb. If you cannot come up with a descriptive name the function probably does too many things.